src/azure-cli/azure/cli/command_modules/reservations/_params.py

```diff
@@ -31,3 +31,17 @@ def load_arguments(self, _):
 
     with self.argument_context('reservations catalog show') as c:
         c.argument('reserved_resource_type', arg_type=get_enum_type(ReservedResourceType))
+
+    with self.argument_context('reservations reservation-order calculate') as c:
+        c.argument('billing_scope_id', options_list=['--scope'])
+        c.argument('reserved_resource_properties', options_list=['--reserved-resource-properties'])
+        c.argument('renew', arg_type=get_three_state_flag())
+        c.argument('reserved_resource_type', arg_type=get_enum_type(ReservedResourceType))
+        c.argument('applied_scope_type', options_list=['--applied-scope-type'], arg_type=get_enum_type(AppliedScopeType))
+
+    with self.argument_context('reservations reservation-order purchase') as c:
```

Comment:
Because there are many parameters similar for `calculate` and `purchase` command, can you abstract them using CLIArgumentType like [here](https://github.com/Azure/azure-cli/blob/8adf97355a605d147afc26f6c1b6f6afb6651c2f/src/azure-cli/azure/cli/command_modules/storage/_params.py#L37)?

Reply:
OK, will do

Reply:
seems in this way I need to put all of those parameters in params by using CLIArgumebtType. Can I keep my original code? I think that looks more clear for me.